### PR TITLE
[build Folder Migration] Move build/Codegen.cmake

### PR DESCRIPTION
**Pull request by jathu**
### Summary

A series of diffs as a part of https://github.com/pytorch/executorch/issues/9117. Move it from `./build` → `./scripts/build`

### Test plan

CI

cc @larryliu0820 @lucylq

### CODE REVIEW

**Review by mergennachin — APPROVED** [review]

**Review by swolchok — APPROVED** [review]

File: backends/cadence/CMakeLists.txt [hunk 1]
```diff
@@ -28,7 +28,7 @@ set(_common_include_directories ${EXECUTORCH_ROOT}/..
 add_compile_definitions(C10_USING_CUSTOM_GENERATED_MACROS)
 
 if(EXECUTORCH_CADENCE_CPU_RUNNER)
-  include(${EXECUTORCH_ROOT}/build/Codegen.cmake)
+  include(${EXECUTORCH_ROOT}/util/build/Codegen.cmake)
```
[the remark this answers is not in the document]
Comment (jathu) [reply backends/cadence/CMakeLists.txt:31]: @swolchok I'm open to suggestions — since this is the first diff, I can easily make the change now

[the remark this answers is not in the document]
Comment (swolchok) [reply backends/cadence/CMakeLists.txt:31]: cmake? scripts? tools?

[the remark this answers is not in the document]
Comment (jathu) [reply backends/cadence/CMakeLists.txt:31]: Not cmake because this contains other files too. Since `scripts` already exists, let's go with that

**Comment by jathu** [comment]
These failing tests are unrelated to this diff:

* [test-arm-backend-delegation / linux-job](https://github.com/pytorch/executorch/actions/runs/13838752311/job/38720474297?pr=9185#logs) — failing in main (lost connection to runner host)
* [test-models-macos (cmake, softmax, portable, macos-m1-stable, 90) / macos-job](https://github.com/pytorch/executorch/actions/runs/13838752311/job/38720586985?pr=9185#logs) — job runner exceeded runtime
